### ISSUE TITLE
QoS_XML results in linking of xercesc with the user code

### ISSUE THREAD

**Issue by jwillemsen** [objectcomputing/OpenDDS#570]
At the moment an application uses the QoS_XML support (as for example the test tests/DCPS/QoS_XML/dump test) there shouldn't be xercesc in the linking of the application. The usage of xercesc should be hidden within the QoS_XML library and shouldn't be exposed to any user code.

**Comment by mitza-oci** [comment]
Wouldn't that break static builds?

**Comment by jwillemsen** [comment]
Yes, if done incorrectly it could. The usage of xercesc should really an internal item, when using shared libraries our dds4ccm connectors shouldn't need to link with xercesc because opendds uses it internally.

**Comment by mitza-oci** [comment]
Can you be more specific than "if done incorrectly"?  I guess the configuration would have to be conditional on static/shared.  Is there an example of this pattern somewhere else within ATCD?

**Comment by ocielliottc** [comment]
There is a feature, static_libs_only, that is set when the -static option is used.  You can define a feature project based on this that can be used to specify different settings when the -static option is used.

ex.,

feature(static_libs_only) {
  libs += foo
}

**Comment by jwillemsen** [comment]
Ok, something to try at some point, thanks @ocielliottc

**Comment by mitza-oci** [comment]
If we use this, also make sure that the configure script uses `-static` with gnuace.  The current version doesn't since the generated makefiles can be used for static or shared.
